### PR TITLE
Provide a way to access all cards in archives/logging screens for admin user (#3145)

### PR DESCRIPTION
**Pull request by vlo-rte**
I decided to name new parameter "adminMode" and not "isAdminMode" in the request to the back to stay coherent with the existing parameter "latestUpdateOnly"... 
 
- In release note :
  -  In chapter : Features
  -  Text : #3145 : Provide a way to access all cards in archives/logging screens for admin user

### CODE REVIEW

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/ArchivedCardCustomRepositoryImpl.java [hunk 1]
```diff
@@ -308,6 +318,17 @@ private boolean isLatestUpdateOnlyParamPresent(MultiValueMap<String, String> par
         return false;
     }
 
+    private boolean isAdminModeParamPresent(CurrentUserWithPerimeters currentUserWithPerimeters,
```
Comment (freddidierRTE) [inline services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/ArchivedCardCustomRepositoryImpl.java:321]: I would name it checkIfInAdminMode as the method is not only checking the param but as well checking the user is admin 
I would put a log with warning level if we want to be in admin mode but we are not member of the admin group

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/UserUtilitiesCommonToCardRepository.java [hunk 2]
```diff
@@ -58,7 +58,11 @@ default Flux<T> findByInitialParentCardUid(ReactiveMongoTemplate template, Strin
     default List<Criteria> computeCriteriaToFindCardByIdWithUser(String id, CurrentUserWithPerimeters currentUserWithPerimeters) {
         List<Criteria> criteria = new ArrayList<>();
         criteria.add(Criteria.where("_id").is(id));
-        criteria.add(computeCriteriaForUser(currentUserWithPerimeters));
+
+        boolean isCurrentUserMemberOfAdminGroup = ((currentUserWithPerimeters.getUserData().getGroups() != null) &&
+                                                   (currentUserWithPerimeters.getUserData().getGroups().contains("ADMIN")));
```
Comment (freddidierRTE) [inline services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/UserUtilitiesCommonToCardRepository.java:63]: i would factorize this method as we need it at least in two places 
adding the method inside CurrentUserWithPerimeters class 
and maybe a unit test

**Comment by freddidierRTE** [comment]
Add karate tests :
- check with admin account either in admin mode or not , i guess in file : src/test/api/karate/cards/fetchArchivedCardsWithParams.feature
- check to get a card in archives with admin even if card is not in perimeter , to check he sees the card

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/ArchivedCardCustomRepositoryImpl.java (hunk 1 above)
Comment (vlo-rte) [reply services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/ArchivedCardCustomRepositoryImpl.java:321]: Done.

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/UserUtilitiesCommonToCardRepository.java (hunk 2 above)
Comment (freddidierRTE) [reply services/cards-consultation/src/main/java/org/opfab/cards/consultation/repositories/UserUtilitiesCommonToCardRepository.java:63]: After discussion not needed yet to factorize

**Comment by vlo-rte** [comment]
> Add karate tests :
> 
>     * check with admin account either in admin mode or not , i guess in file : src/test/api/karate/cards/fetchArchivedCardsWithParams.feature

Done.

**Pull request merged**